### PR TITLE
New Date and Time picker

### LINKED ISSUE #459: Need better guidance on selecting date and time
#### 1. Choose the type of contribution for this pattern:
- [ x] This is a new pattern 
- [ ] This is an enhancement on an existing pattern
#### 4. Use Cases + Requirements
* User wants to set a date, time, or date and time.
* User wants to set a date range for filtering.
We currently have Datepicker and Timepicker widgets but no guidance on how to use these in common use cases.
#### 5. References 
None
#### 6. Design Concepts  
None
#### 7. Scope  
Research and design
#### 9. Pattern Name + Location  
Propose "Date and Time" to be added to forms and controls.

### PR DESCRIPTION
**Pull request by nding-anges**
## Description
Fixes #459 
* Update currently Date picker in From&Control as well as add new variation pickers to meet new requirements . All variation pickers summarized as below:

1. DatePicker
2. TimePicker
3. DateTimePicker
4. DateRangePicker
5. DateTimeRangePicker

* Update and add new design guidance of all variation pickers of  "Date and Time Picker" 

## Changes

* Update Overview.md
* Update design.md
* Update site.md
* Change name to Date and Time
* Add some images of components
* Add new guidance for all components

## Contribution Submission Checklist

- [ ] A GitHub issue was created and reviewed with the PatternFly community.
- [ ] The pattern design overview.md and design.md templates were used to provide all information, relevant to this design.
- [ ] Any new files have been added to the folder for that pattern.
- [ ] The folder, for brand new patterns, has been titled with the name of the pattern.
- [x] All images are in an "img" folder and embedded images have the correct filepath.
- [x] The overview.md, design.md and site.md files, for this design have been commited and pushed to the patternfly-design repository, for review.

### CODE REVIEW

**Comment by mcarrano** [comment]
@nding-anges I see that you closed this and I assume that was because there were some bugs in your design.md files. But since I did look at this, I wanted to just give feedback on a couple of things as you work on a revised version.

* I'd like to rename this to just Date and Time. That means also changing the folder name and updating the site.md since the name that shows up on the website is based on these two things.

* I think that the design is good, but I was hoping that we can revise the Overview page to focus more on what the use cases are and what solution is best for the particular use case. Think of this as providing guidance to designers. As a designer, what should they know about working with date and time in PatternFly? 

Does this make sense? If we need to discuss, would be glad to set up a time to meet.

**Comment by nding-anges** [comment]
@mcarrano Thanks a lot! I'll modifier overview page. I just thinks about what should be some particular use cases? Could you please clarify it or make an example to me ? And in the meantime, we still need to confirm which alternative design of DateTime picker should be take.See detail here:https://docs.google.com/document/d/1kVAhNzSB-szucfeGgq815dB6e214WavZ-gqRR0Sydj0/edit?disco=AAAAB6F0jA4

**Comment by mcarrano** [comment]
@nding-anges I think some of the common use cases are:

* I want to set a single date or a single time.
* I want to set single date and time in combination.
* I want to filter or search for items by a range of dates or date and time in combination.

Can you think of any others? I don't think this is vastly different from what you already have, but it emphasizes the use rather than just what the component is called. Does that help?

I also reviewed and commented on the Google Doc. Thanks for creating additional options.

**Comment by nding-anges** [comment]
@mcarrano I get your point.Thanks a lot!  I just made some update of Overview part in order to emphasizes the use case. Please have a look.

File: pattern-library/forms-and-controls/datepicker/design/overview.md [hunk 1]
```diff
@@ -1,7 +1,42 @@
...
+# Date and Time
 
-The datepicker allows a user to select a single date. The most common use case is selecting a date for something to run or expire. The datepicker widget is based on the Bootstrap-datepicker. See the [bootstrap-datepicker doc](http://bootstrap-datepicker.readthedocs.io/en/latest/) for complete documentation.
+The date and  time picker allows a user to select a single point in time or a time range. Time can be expressed as a date, a time of day, or a combination of both. Common use cases are selecting a date and time for something to run or expire and filtering events by their timestamp. The date and time picker is based on the Bootstrap-datetimepicker. See the
+[bootstrap-datepicker doc](http://http://eonasdan.github.io/bootstrap-datetimepicker/) for complete documentation.
 
-This pattern should NOT be used when selecting a date range.
+There are number of variation of the date and Time picker. We recommend using appropriate component to meet specific user case, jump to a section below for applied examples:
+</br>
+* [As a user, I want to set a single date](#DatePicker)
```
Comment (mcarrano) [inline pattern-library/forms-and-controls/datepicker/design/overview.md]: These jump links don't seem to be working in GitHub preview. Not sure why, but can you check them?

File: pattern-library/forms-and-controls/datepicker/design/design.md [hunk 2]
```diff
@@ -1,15 +1,94 @@
-# Datepicker
+# Date and Time
+Date and Time picker has a number of variations, jump to a section below for applied examples:
+* [DatePicker](#DatePicker)
```
Comment (mcarrano) [inline pattern-library/forms-and-controls/datepicker/design/design.md]: These jump links do not seem to be working in GitHub preview, not sure why.

Comment (junezhang) [reply pattern-library/forms-and-controls/datepicker/design/design.md]: @anges-ding, please refrence the link for how to make link on the same page in Markdown.
https://stackoverflow.com/questions/6695439/how-to-link-to-a-named-anchor-in-multimarkdown

. update your third title to sencond title
. make the #id is lower-case.

File: pattern-library/forms-and-controls/datepicker/design/overview.md (hunk 1 above)
Comment (junezhang) [reply pattern-library/forms-and-controls/datepicker/design/overview.md]: @anges-ding, these are the user cases, where we proposed to link? Could we remove the link?

Comment (junezhang) [reply pattern-library/forms-and-controls/datepicker/design/overview.md]: Same with my last comment, update #DatePicker to #datepicker, the name should be a lower-case

File: pattern-library/forms-and-controls/date-and-time/site.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+title: Date and Time
+layout: page-pattern
+overview: pattern-library/forms-and-controls/datepicker/design/overview.md
```
Comment (mcarrano) [inline pattern-library/forms-and-controls/date-and-time/site.md]: The path names for the Overview and Design pages must be updated here to reflect the new folder location if these pages are to be displayed.

**Comment by mcarrano** [comment]
I am going to merge this.  If there are further comments or issues, we can follow up with a new PR. Thanks @nding-anges for your efforts in getting this finalized.

**Comment by nding-anges** [comment]
@mcarrano Thanks!

**Pull request merged**